### PR TITLE
Utilities: Add pathchk

### PR DESCRIPTION
**Pull request by bcoles**
Adds [pathchk](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/pathchk.html) - one of the remaining few missing POSIX utilities.

Test script:

```
#!/bin/sh
pathchk '/doesnotexist'
pathchk '/root/a'
pathchk --portability '/a/@'
pathchk --portability 'a'
pathchk --portability '/a/'
pathchk --portability '/-a/'
pathchk --portability ''
pathchk --portability '/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
pathchk --portability '/a/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA'
pathchk '/a/AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA'
```

pathchk

### CODE REVIEW

**Comment by bgianfo** [comment]
Consider adding a man page to go along with the utility?

File: Userland/Utilities/pathchk.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    if (flag_portability) {
+        flag_most_posix = true;
+        flag_empty_name_and_leading_dash = true;
+    }
+
+    for (auto& path : paths) {
+        unsigned long path_max = flag_most_posix ? _POSIX_PATH_MAX : pathconf(path, _PC_PATH_MAX);
+        unsigned long name_max = flag_most_posix ? _POSIX_NAME_MAX : pathconf(path, _PC_NAME_MAX);
+
+        if (String(path).length() > path_max) {
```
Comment (bgianfo) [inline Userland/Utilities/pathchk.cpp:63]: You are converting path to String in multiple places, do it once at the beginning of the loop?
Also does it really need to be a string, could we get away with StringView?

**Comment by bcoles** [comment]
>Consider adding a man page to go along with the utility?

nah

File: Userland/Utilities/pathchk.cpp (hunk 1 above)
Comment (bcoles) [reply Userland/Utilities/pathchk.cpp]: > You are converting path to `String` in multiple places, do it once at the beginning of the loop?

Done.

> Also does it really need to be a `String`, could we get away with `StringView`?

Sometimes it needs to be a String.

**Pull request merged**